ci/check_enterprise_merge.sh

```diff
@@ -55,12 +55,16 @@ git checkout "enterprise/enterprise-master"
 if git merge --no-commit "origin/$PR_BRANCH"; then
     echo "INFO: community PR branch could be merged into enterprise-master"
     # check that we can compile after the merge
-    check_compile
-    exit 0
+    if check_compile; then
+        exit 0
+    fi
+
+    echo "WARN: Failed to compile after community PR branch was merged into enterprise"
+else
+    # undo partial merge
+    git merge --abort
```

Comment:
I think this should be done always. i.e. it should start with a clean slate when trying to merge the enterprise PR branch.

Reply:
if the merge was successful then there won't be any partial merge and `git merge --abort` will return a non-zero error code, so shouldn't this be only done when there is partial merge?

Reply:
I mean we could do `git merge --abort || true` but we have pipefail set.